src/hardware/disney.cpp

```diff
@@ -27,49 +27,52 @@
 #include "pic.h"
 #include "setup.h"
 
-#define DISNEY_BASE 0x0378
+// Disney Sound Source Constants
+constexpr uint16_t DS_PORT_BASE = 0x0378;
+constexpr uint8_t DS_BUF_SAMPLES = 128;
 
-#define DISNEY_SIZE 128
+constexpr uint8_t DS_ACKNOWLEDGE_BIT = 0b0100'0000;  // 0x40
+constexpr uint8_t DS_INTERRUPT_MASK = 0b1111'1011;   // ~0x4
+constexpr uint8_t DS_INIT_STATUS_BITS = 0b1000'0100; // 0x84
+constexpr uint8_t DS_PARALLEL_IRQ_BIT = 0b0001'0000; // 0x10
+constexpr uint8_t DS_PIN_9_BIT = 0b1000'0000;        // 0x80
 
-constexpr uint8_t DISNEY_INIT_STATUS = 0b1000'0100; // 0x84
+enum DS_STATE { IDLE, RUNNING, FINISH, ANALYZING };
 
 typedef struct _dac_channel {
-	Bit8u buffer[DISNEY_SIZE];	// data buffer
-	Bitu used;					// current data buffer level
-	double speedcheck_sum;
-	double speedcheck_last;
-	bool speedcheck_failed;
-	bool speedcheck_init;
+	uint8_t buffer[DS_BUF_SAMPLES]; // data buffer
+	uint8_t used = 0;               // current data buffer level
+	double speedcheck_sum = 0;
+	double speedcheck_last = 0;
+	bool speedcheck_failed = false;
+	bool speedcheck_init = false;
 } dac_channel;
 
 using mixer_channel_ptr_t = std::unique_ptr<MixerChannel, decltype(&MIXER_DelChannel)>;
+
 static struct {
 	IO_ReadHandleObject read_handler{};
 	IO_WriteHandleObject write_handler{};
 
 	// parallel port stuff
-	Bit8u data;
-	uint8_t status = DISNEY_INIT_STATUS;
+	uint8_t data = 0;
+	uint8_t status = DS_INIT_STATUS_BITS;
```

Comment:
Are these bits or a number? I suspect old name was fine in here…

Reply:
See prior comment. The initial state is composed if two states set high. I'm unable to figure them out yet; maybe you can help?

For example, maybe 0x80 represents stereo mode, and 0x04 irq high.

Once these are identified, we would use them instead:

DS_INITIAL_STATE = STEREO_BIT | IRQ_BIT